src/cli/self_update/windows.rs

```diff
@@ -285,6 +282,75 @@ pub fn do_remove_from_path() -> Result<()> {
     _apply_new_path(new_path)
 }
 
+pub fn do_add_to_programs() -> Result<()> {
+    use std::path::PathBuf;
+
+    let key = RegKey::predef(HKEY_CURRENT_USER)
+        .create_subkey(r"Software\Microsoft\Windows\CurrentVersion\Uninstall\rustup")
+        .chain_err(|| ErrorKind::PermissionDenied)?
+        .0;
+
+    // Don't overwrite registry if rustup is already installed
+    let prev = key
+        .get_raw_value("UninstallString")
+        .map(|val| from_winreg_value(&val));
+    if let Ok(Some(s)) = prev {
+        let mut path = PathBuf::from(OsString::from_wide(&s));
+        path.pop();
+        if path.exists() {
+            return Ok(());
+        }
+    }
+
+    let mut path = utils::cargo_home()?;
+    path.push("bin\rustup.exe");
+    let mut uninstall_cmd = path.into_os_string();
+    uninstall_cmd.push(" self uninstall");
+
+    let reg_value = RegValue {
+        bytes: to_winreg_bytes(uninstall_cmd.encode_wide().collect()),
+        vtype: RegType::REG_SZ,
+    };
+
+    key.set_raw_value("UninstallString", &reg_value)
+        .chain_err(|| ErrorKind::PermissionDenied)?;
+    key.set_value("DisplayName", &"rustup - Rust toolchain manager")
+        .chain_err(|| ErrorKind::PermissionDenied)?;
+
+    Ok(())
+}
+
+pub fn do_remove_from_programs() -> Result<()> {
+    use std::io;
+    use std::path::PathBuf;
+
+    let root = RegKey::predef(HKEY_CURRENT_USER);
+
+    let key = match root.open_subkey(r"Software\Microsoft\Windows\CurrentVersion\Uninstall\rustup")
+    {
+        Ok(key) => key,
+        Err(ref e) if e.kind() == io::ErrorKind::NotFound => return Ok(()),
+        Err(e) => return Err(e).chain_err(|| ErrorKind::PermissionDenied),
+    };
+
+    // Don't remove uninstall information from another installation
```

Comment:
I think we should remove this: if someone has moved their homedir around for whatever reason, this logic will make it impossible to remove the entry from their programs list - and thats super annoying.

OTOH removing an entry thats not for a parallel install done using CARGO_HOME + RUSTUP_HOME variables, is somewhat harmless, since the user can still use rustup, and still uninstall directly, as long as we document that.